### PR TITLE
feat: implement SupportDevelopmentAsync() function

### PR DESCRIPTION
**Pull request by ValeraFinebits**
## Issue
N/A

## Description
Implement SupportDevelopmentAsync() function

## Pull Request Checklist

Please check if your pull request fulfills the following requirements:

- [x] The pull request is named according to the format: _[\<type\>](#types-of-pull-requests)(optional scope): \<description\>_
- [ ] The pull request is associated with a GitHub issue. Note: if possible use the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword)
- [x] [The pull request is labeled](#pull-request-labels)
- [x] [Affected platforms are labeled](#platform-labels)
- [ ] Commits follow the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)
- [x] There are no difficult to understand places left without comments
- [x] The changes do not generate new warnings

> [!NOTE]
>
> <details><summary><strong id="types-of-pull-requests">Types of pull requests</strong></summary>
>
> - `feat` - adding **new features**
> - `fix` - **bug** fixes
> - `test` - adding or correcting **tests**
> - `perf` - changes that improve **performance**
> - `refactor` - simple **rewriting** or **restructuring** of code without adding new features or fixing bugs
> - `style` - changes in **code styles** and no changes in logic
> - `build` - changes related to **the build of the project** and **dependencies**
> - `ci` - changes related to **continuous integration**
> - `docs` - changes in **documentation** or just **comments** in source code
> - `chore` - something that **doesn't fit** the other possible types
>
> </details>
>
> <details><summary><strong id="pull-request-labels">Pull request labels</strong></summary>
>
> - **`type/breaking-change`** - pull requests with changes that are **not backward compatible**
> - `type/build` - pull requests that change the **project's build** or **dependencies**
> - `type/chore` - pull requests **without** making **changes** to the code, project build, formatting, documentation, etc
> - `type/ci` - pull requests whose changes are related to **continuous integration**
> - `type/documentation` - pull requests that only change **documentation**
> - **`type/feature`** - pull requests that add **new features**
> - **`type/fix`** - pull requests that fix a **bug**
> - **`type/localization`** - pull requests that change **translation**
> - **`type/performance`** - pull requests that improve **performance**
> - `type/refactor` - pull requests that **refactor** a section of code
> - `type/style` - pull requests that change **code styles**
> - `type/test` - pull requests that add or correct **tests**
> - **`ignore-for-release`** - for pull requests that do **not need** to be appeared in **release notes**
>
> ---
> A pull request appears in the release notes if it has one of the labels: `type/breaking-change`, `type/feature`, `type/fix`, `type/localization`, `type/performance`
>
> </details>
>
> <details><summary><strong id="platform-labels">Affected platforms labels</strong></summary>
>
> - `platform/all` - pull requests that are related to the all platforms
> - `platform/android` - pull requests that are related to the Android platform
> - `platform/desktop` - pull requests that are related to the desktop
> - `platform/ios` - pull requests that are related to the iOS platform
> - `platform/macos` - pull requests that are related to the macOS platform
> - `platform/other` - pull requests that are related to an unknown platform
> - `platform/uwp` - pull requests that are related to the Universal Windows Platform
> - `platform/wasm` - pull requests that are related to the WebAssembly platform
> - `platform/winui` - pull requests that are related to the WinUI platform
>
> </details>
>

### CODE REVIEW

File: source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using Tuvi.App.ViewModels.Services;
+using Windows.Services.Store;
+
+namespace Tuvi.App.Shared.Services
+{
+    public class PurchaseService : IPurchaseService
+    {
+        public Task BuySubscriptionAsync()
```
Comment (al-kau) [inline source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs:12]: Maybe the method should be like `public Task<bool> BuySubscriptionAsync()` ?

File: source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        {
+            var subscription = new SubscriptionProduct("<InAppOfferToken>", "<ProductId>");
+            return subscription.PurchaseAsync();
+        }
+    }
+
+    public class SubscriptionProduct
+    {
+        private string InAppOfferToken { get; set; }
```
Comment (al-kau) [inline source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs]: `set;` can be removed

File: source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        private string ProductId { get; set; }
```
Comment (al-kau) [inline source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs]: `set;` can be removed

File: source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+            return _storeContext;
+        }
+
+        public async Task<bool> PurchaseAsync()
+        {
+            StoreContext context = StoreContext.GetDefault();
+
+            var id = ProductId;
+
+            string[] productKinds = { "Durable" };
```
Comment (al-kau) [inline source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs]: Make a constant for `"Durable"`

File: source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs (hunk 1 above)
Comment (ValeraFinebits) [reply source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs:12]: Done

File: source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs (hunk 2 above)
Comment (ValeraFinebits) [reply source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs]: Done

File: source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs (hunk 3 above)
Comment (ValeraFinebits) [reply source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs]: Done

File: source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs (hunk 4 above)
Comment (ValeraFinebits) [reply source/Eppie.App/Eppie.App.Shared/Services/PurchaseService.cs]: Done

**Review by al-kau — APPROVED** [review]

**Pull request merged**